core/common/boot/src/main/java/org/eclipse/edc/boot/system/runtime/BaseRuntime.java

```diff
@@ -135,6 +142,27 @@ protected ServiceExtensionContext createServiceExtensionContext(Config config) {
         return context;
     }
 
+    /**
+     * Sets the ConsoleMonitor level from config, if specified. The config has precedence over the program argument.
+     */
+    @NotNull
+    protected String[] setLogLevelProgArgFromConfig(Config config, Monitor monitor, String[] programArgs) {
+        if (monitor instanceof ConsoleMonitor) {
```

Comment:
the fact aside that the `BaseRuntime` should not know anything about a specific monitor, why is the log level only relevant for a `ConsoleMonitor`?

Reply:
It's not. But the `Monitor` interface doesn't declare a level attribute which in my opinion leaves this specificity to the implementations.
The reason why this is here is because I couldn't think of another option to load the log level from the config.

Reply:
To be honest I don't think log level should be set with configuration, there's a clear reason why it gets set with program arguments. I would remove the if branch at lines 91:94 and this protected method as well

Reply:
Agreed, the design of the Monitor is that the host (runtime bootstrap) decides how to source and configure it. In the default case, this is done by the bootstrapper and should never be sourced from configuration.

As I mentioned, this PR can be simplified to only modify one class.